### PR TITLE
Add limit for graph connections and data size

### PR DESCRIPTION
**Pull request by Linary**

### CODE REVIEW

File: hubble-be/src/main/java/com/baidu/hugegraph/service/LicenseService.java [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        long actualDataSize = getActualDataSize(client, conn.getGraph());
+        if (allowedDataSize != Constant.NO_LIMIT &&
+            actualDataSize > allowedDataSize) {
+            String msg = this.getMessage("license.verify.datasize.exceed",
+                                         conn.getName(), actualDataSize,
+                                         allowedDataSize);
+            conn.setEnabled(false);
+            conn.setDisableReason(msg);
+            this.connService.update(conn);
+        }
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/service/LicenseService.java]: add else

File: hubble-be/src/main/java/com/baidu/hugegraph/service/LicenseService.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+            msg = connection.getDisableReason();
+        }
+        if (msg != null) {
+            throw new ExternalException(Constant.STATUS_UNAUTHORIZED, msg);
+        }
+    }
+
+    @Async
+    @Scheduled(fixedDelay = 10 * 60 * 1000)
+    public void updateAllGraphStatus() {
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/service/LicenseService.java]: test if throw exception

File: hubble-be/src/main/java/com/baidu/hugegraph/license/LicenseVerifier.java [hunk 3]
```diff
@@ -75,6 +85,10 @@ public static LicenseVerifier instance() {
         return INSTANCE;
     }
 
+    public String edition() {
+        return this.edition;
+    }
+
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/license/LicenseVerifier.java]: update allowedGraphs():
return this.manager.allowedGraphs();

Comment (javeme) [reply hubble-be/src/main/java/com/baidu/hugegraph/license/LicenseVerifier.java]: address it

File: hubble-be/src/main/java/com/baidu/hugegraph/advisor/MessageSourceHandler.java [hunk 4]
```diff
@@ -43,34 +44,42 @@
...
         }
     }
 
     private Locale getLocale() {
-        Locale locale;
+        try {
+            RequestContextHolder.currentRequestAttributes();
+        } catch (IllegalStateException e) {
+            return DEFAULT_LOCALE;
+//            return LocaleContextHolder.getLocale();
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/advisor/MessageSourceHandler.java]: unused?

File: hubble-be/src/main/java/com/baidu/hugegraph/advisor/MessageSourceHandler.java [hunk 5]
```diff
@@ -43,34 +44,42 @@
...
         if (userInfo != null && userInfo.getLocale() != null) {
-            locale = LocaleUtils.toLocale(userInfo.getLocale());
+            return LocaleUtils.toLocale(userInfo.getLocale());
         } else if (this.request.getLocale() != null) {
-            locale = RequestContextUtils.getLocale(this.request);
+            return RequestContextUtils.getLocale(this.request);
         } else {
-            locale = LocaleContextHolder.getLocale();
+            return DEFAULT_LOCALE;
+//            return LocaleContextHolder.getLocale();
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/advisor/MessageSourceHandler.java]: ditto

File: hubble-be/src/main/java/com/baidu/hugegraph/exception/ExternalException.java [hunk 6]
```diff
@@ -21,11 +21,28 @@
...
 
+    private int status;
+
     public ExternalException(String message, Object... args) {
+        this(400, message, args);
+    }
+
+    public ExternalException(int status, String message, Object... args) {
         super(message, args);
+        this.status = status;
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/exception/ExternalException.java]: also call this(...)

File: hubble-be/src/main/resources/i18n/messages_zh_CN.properties [hunk 7]
```diff
@@ -102,14 +106,18 @@ execute.predication.error=执行判断语句出错
...
 license.install.failed=安装证书失败
 license.public-cert.invalid=非法的公钥文件
 license.read.public-cert.failed=读取公钥信息失败
 license.read.failed=读取证书信息失败
 license.verify.failed=验证证书失败
 license.verify.server-id.unmatch=hugegraph-hubble 服务的 id {0} 不在已授权范围内
 license.verify.version.unmatch=hugegraph-hubble 的当前版本 {0} 不在已授权范围内 <= {1}
-license.verify.graphs.exceed=图连接个数 {0} 超过证书限制 {1}
+license.verify.graphs.exceed=图连接个数 {0} 超过社区版上限 {1} 个
+license.verify.datasize.exceed=图 {0} 已占用磁盘 {1}MB 超过社区版上限 {2}MB
```
Comment (javeme) [inline hubble-be/src/main/resources/i18n/messages_zh_CN.properties]: add one more param for edition

Comment (Linary) [reply hubble-be/src/main/resources/i18n/messages_zh_CN.properties]: The message key nested is too complicated, I think remove edition is better

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java [hunk 8]
```diff
@@ -58,8 +58,12 @@
 public class EdgeLabelController extends SchemaController {
 
     private static final List<String> PRESET_COLORS = ImmutableList.of(
-            "#5C73E6", "#569380", "#8ECC93", "#F79767", "#F06667",
-            "#C990C0", "#4D8DDA", "#57C7E3", "#FFE081", "#DA7194"
+            // old
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java]: bright colors

File: hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java [hunk 9]
```diff
@@ -58,8 +58,12 @@
...
+            "#2B65FF", "#0EB880", "#76C100", "#ED7600", "#E65055",
+            "#A64EE6", "#108CEE", "#00B5D9", "#F2CA00", "#E048AE",
+            // new
```
Comment (javeme) [inline hubble-be/src/main/java/com/baidu/hugegraph/controller/schema/EdgeLabelController.java]: dull color

**Pull request merged**